### PR TITLE
refactor(angular): angular component metadata generation

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Description

Add a short description of:
- Instead of having values in string moved them to Object
- Now it is in Object and can be more or less better overridden by the user
- Also, touched a bit `imports` generation

### CODE REVIEW

**Review by samijaber — APPROVED** [review]
Thanks for this!

**Comment by samijaber** [comment]
you'll need to run `fmt:prettier` at the repo root to fix your PR

**Comment by g-cheishvili** [comment]
Yeah, noticed. I am off from PC now, its 2:30 am here, will do tomorrow

File: packages/core/src/generators/angular.ts [hunk 1]
```diff
@@ -335,7 +335,25 @@ export const componentToAngular: TranspilerGenerator<ToAngularOptions> =
...
         }),
     });
-
+    // Preparing built in component metadata parameters
+    const componentMetadata: Record<string, any> = {
+      selector: `'${kebabCase(json.name || 'my-component')}'`,
+      template: `\`
+        ${indent(template, 8).replace(/`/g, '\\`').replace(/\$\{/g, '\\${')}
+        \``,
+      ...(css.length ? { styles: `\`${indent(css, 8)}\`` } : {}),
```
Comment (samijaber) [inline packages/core/src/generators/angular.ts:344]: You removed the `[]` brackets in this string, this breaks the rendering of the styles property. you can see that the snapshot tests are all shifted.

```suggestion
      ...(css.length ? { styles: `[\`${indent(css, 8)}\`]` } : {}),
```

Make sure to run `yarn snapupdate` in `packages/core` locally to update test snapshots and see if there are changes there

**Review by samijaber — CHANGES_REQUESTED** [review]

**Review by samijaber — APPROVED** [review]
Awesome, welcome to Mitosis! ⚡

**Pull request merged**